Management/Clients/AirQualityDataClient.cs

```diff
@@ -45,7 +54,101 @@ public async Task<AirQualityCityResponse> GetAirQualityAsync(string city, string
             throw new Exception("Retrieve Data Failed");
         }
 
-        public int CalculateScoreAsync()
-            => throw new NotImplementedException();
+        // Use Air Quality Index
+        // https://www.airnow.gov/aqi/aqi-basics/
+        public async Task<Dictionary<City, int>> CalculateScoresAsync(
+            IEnumerable<City> cities,
+            State state,
+            CountryCode countryCode,
+            CancellationToken cancellationToken)
+        {
+            var bag = new ConcurrentBag<(City city, int score)>();
+
+            var tasks = cities.Select(async city =>
+            {
+                var response = await GetSingleCityAsync(city, state, countryCode, cancellationToken);
+                bag.Add(response);
+            });
+
+            await Task.WhenAll(tasks);
+            var count = bag.Count;
+
+            return bag
+                .ToLookup(b => b.city)
```

Comment:
Why I do this?
To prevent duplicate key insert into dictionary exception.
To lookup will group result with same key.
We should not have duplicate entry base on the logic, but we should never trust the incoming data!